momento-sdk/src/main/java/momento/sdk/SubscriptionWrapper.java

```diff
@@ -19,42 +21,54 @@
 public class SubscriptionWrapper implements Closeable {
   private final Logger logger = LoggerFactory.getLogger(SubscriptionWrapper.class);
   private final ScsTopicGrpcStubsManager grpcManager;
-  private final String cacheName;
-  private final String topicName;
-  private final SubscriptionState subscriptionState;
-  private final ISubscriptionCallbacks options;
-  private StreamObserver<_SubscriptionItem> subscription;
-
-  public SubscriptionWrapper(
-      ScsTopicGrpcStubsManager grpcManager,
-      String cacheName,
-      String topicName,
-      SubscriptionState subscriptionState,
-      ISubscriptionCallbacks options) {
+  private final SendSubscribeOptions options;
+  private ScheduledExecutorService scheduler;
+
+  private CancelableClientCallStreamObserver<_SubscriptionItem> subscription;
+
+  public SubscriptionWrapper(ScsTopicGrpcStubsManager grpcManager, SendSubscribeOptions options) {
     this.grpcManager = grpcManager;
-    this.cacheName = cacheName;
-    this.topicName = topicName;
-    this.subscriptionState = subscriptionState;
     this.options = options;
   }
 
-  public CompletableFuture<Void> subscribe() {
+  public CompletableFuture<Void> subscribeWithRetry() {
     CompletableFuture<Void> future = new CompletableFuture<>();
+    subscribeWithRetryInternal(future);
+    return future;
+  }
 
+  public void subscribeWithRetryInternal(CompletableFuture<Void> future) {
     subscription =
-        new StreamObserver<_SubscriptionItem>() {
+        new CancelableClientCallStreamObserver<_SubscriptionItem>() {
           boolean firstMessage = true;
 
+          @Override
+          public boolean isReady() {
+            return false;
+          }
+
+          @Override
+          public void setOnReadyHandler(Runnable onReadyHandler) {}
+
+          @Override
+          public void request(int count) {}
+
+          @Override
+          public void setMessageCompression(boolean enable) {}
+
+          @Override
+          public void disableAutoInboundFlowControl() {}
```

Comment:
can you push these overrides up into your `CancelableClientCallStreamObserver` class, so that consumers of it don't have to worry about them?